build.go

```diff
@@ -139,21 +143,56 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		return err
 	}
 
-	return c.lifecycle.Execute(ctx, build.LifecycleOptions{
+	lifecycleOpts := build.LifecycleOptions{
 		AppPath:            appPath,
 		Image:              imageRef,
 		Builder:            ephemeralBuilder,
 		RunImage:           runImageName,
 		ClearCache:         opts.ClearCache,
 		Publish:            opts.Publish,
+		TrustBuilder:       opts.TrustBuilder,
```

Comment:
The code below might be made a bit simpler (and potentially more unit-testable) by introducing the concept of `build.LifecycleOptions.UseCreator`. Thoughts?